### DISCUSSION TITLE
Always expose port 80

### DISCUSSION THREAD

**Discussion by vilyansky** [tiangolo/meinheld-gunicorn-flask-docker#158] [Questions] [answered]
Image always expose port 80 regardless PORT, BIND environment variables settings.
Steps to reproduce:
1. Adds to Dockerfile

> FROM tiangolo/meinheld-gunicorn-flask:python3.7-alpine3.8
> ENV BIND 0.0.0.0:8080
> ENV PORT 8080
> ENV LOG_LEVEL debug
> EXPOSE 8080:8080

2. build & run image
3. docker ps -a

> CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS               NAMES
> c36edb38d913        someuser/someimage:latest   "/entrypoint.sh /sta…"   9 minutes ago       Up 9 minutes        80/tcp, 8080/tcp    stupefied_morse

**Comment by BoforsKing** [comment]
Have you tried setting these on your dockerfile:

```dockerfile
ENV LISTEN_PORT 5000
EXPOSE 5000
```

**Comment by vilyansky** [comment]
Yes. It have a same result.
Cause of this that the base image have
`
EXPOSE 80
`
in it's Dockerfile - https://hub.docker.com/r/tiangolo/meinheld-gunicorn/dockerfile

**Comment by BoforsKing** [comment]
I just checked some internal project dockerfile and noticed I am not using `meinheld-gunicorn-flask` but `meinheld-gunicorn:python3.7`.

You can give it a try.

**Comment by vilyansky** [comment]
It will have same result. The image have expose default http port that can be unused in next stages. It's no critical but add misleads to containers list.
I can build my custom image of course. 
Everything must be DIY, yeah.

**Comment by fabiom91** [comment]
So what's the solution?

**Comment by sysmetic** [comment]
docker run -p 3004:3004 -e PORT="3004" -e MODULE_NAME="app" myimage

it works.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :nerd_face: 

I just removed `EXPOSE 80` in the base image for this (https://github.com/tiangolo/meinheld-gunicorn-docker/pull/128). :rocket:
